torch/csrc/autograd/FunctionsManual.cpp

```diff
@@ -1888,9 +1888,18 @@ Tensor kl_div_target_backward(
     Tensor target,
     int64_t reduction,
     bool log_target) {
+  // No in-place operations on target unless it is not broadcasted
+  const auto is_broadcasted_target = [&]() {
+    const auto target_storage = target.storage();
+    const auto target_dtype_nbytes = c10::elementSize(target.scalar_type());
+    const auto target_storage_numel = target_storage.nbytes() / target_dtype_nbytes;
+    return target_storage_numel == target.numel();
+  }();
```

Comment:
I think we would still miss cases like  `a = torch.tensor([1., 2., 3., 4.])[0].expand(2, 2)` here?

Not completely sure if memory overlap is the issue either since performing in-place on it already produces an error:
```
>>> import torch
>>> torch.tensor([1.]).expand(2, 2)
tensor([[1., 1.],
        [1., 1.]])
>>> a = torch.tensor([1.]).expand(2, 2)
>>> a += 1
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
RuntimeError: unsupported operation: more than one element of the written-to tensor refers to a single memory location. Please clone() the tensor before performing the operation.
```

Reply:
Shall I remove it? I could just restrict to using out of place operations between `target` and `self`.